stixpy/product/sources/science.py

```diff
@@ -516,14 +555,117 @@ def times(self):
         return self.data["time"]
 
     @property
-    def durtaion(self):
+    def duration(self):
         """
         An `astropy.units.Quantiy` array giving the duration or integration time
         """
         return self.data["timedel"]
 
+    def crop_by_value(
+        self,
+        values,
+    ):
+        r"""
+        Crop the data by a values e.g time, energy.
+
+        Parameters
+        ----------
+        values
+
+        Returns
+        -------
+
+        Examples
+        --------
+        >>> cpd.crop_by_value()
+
+        """
+
+    def rebin(self, *indices, operator: Callable = np.sum):
+        r"""
+        Rebin the data using the given indices.
+
+        Parameters
+        ----------
+        operator :
+        indices :
+
+        Returns
+        -------
+
+        Examples
+        --------
+        cpd.rebin(None, None, None, [1, 5])
```

Comment:
I think it would be good for this to follow the `NDCube.rebin` API as much as makes sense.  In that case, `rebin` should take:
* integers which are an integer factor of the length of the axis to which it corresponds, or:
* a sequence of indices representing the bin edges along that axis.

In this case, the equivalent API for no rebinning should be `1`, not `None`.

Alternatively, this method could be renamed to avoid confusion with `NDCube.rebin` to something like `rebin_by_edges`.  In that case, it could take `None` or a sequence of indices for each axis.